daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/GroupDef.scala

```diff
@@ -131,6 +133,18 @@ sealed abstract class GlobalGroupDef(
     }
   }
 
+  private def checkForGroupDefAnnotations(): Unit = {
+    // Ensure that the group definition itself does not have any annotations.
+    // Annotations should only be on group references or children of the group
+    // definition
+    val dais = getDFDLAppinfos(defXML \ "annotation" \ "appinfo")
+    if (dais.nonEmpty)
+      SDW(
+        WarnID.InvalidAnnotationPoint,
+        "Annotations placed directly on a group definition will be ignored by DFDL. Any annotation expected to be processed by DFDL should instead be placed on the group reference",
```

Comment:
As an alternative to the group reference, could annotations also be placed on the model group (i.e. choice or sequence) of the group definition? I imagine if someone put an annotation on the group def that's maybe what they meant to do? Is it worth saying something like `... should be placed on the group reference or model group`?

Reply:
Saying "or model group" may confuse a user who isn't familiar with the meaning of "model group."  I suggest either leaving the message the way it is now or changing it to `... should be placed on the group reference, choice, or sequence` if we really want to list all the possible places.